mne/epochs.py

```diff
@@ -2529,7 +2562,7 @@ def to_data_frame(
         time_format = _check_time_format(time_format, valid_time_formats)
         # get data
         picks = _picks_to_idx(self.info, picks, "all", exclude=())
-        data = self.get_data()[:, picks, :]
+        data = self._get_data(on_empty="raise")[:, picks, :]
```

Comment:
I was reluctant to change the signature of public methods, so I sometimes switched to the private method. WDYT?